IntelliTectAnalyzer/IntelliTectAnalyzer/DescriptionAttribute.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+
+namespace IntelliTectAnalyzer
+{
+    [AttributeUsage(AttributeTargets.All)]
+    internal class DescriptionAttribute : Attribute
+    {
+        public static DescriptionAttribute Default { get; } = new DescriptionAttribute();
+        
+        public string Description { get; }
+
+        public DescriptionAttribute() : this(string.Empty)
+        {
+        }
+
+        public DescriptionAttribute(string description)
+        {
+            Description = description;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == this)
```

Comment:
This works, but because the == operator can be overloaded it would be better to use reference equality.
```suggestion
            if (ReferenceEquals(obj, this))
```